src/IrcTransport/IrcWebSocketTransport.cs

```diff
@@ -0,0 +1,169 @@
+﻿/*
+ * $Id$
+ * $URL$
+ * $Rev$
+ * $Author$
+ * $Date$
+ *
+ * SmartIrc4net - the IRC library for .NET/C# <http://smartirc4net.sf.net>
+ *
+ * Copyright (c) 2003-2009 Mirco Bauer <meebey@meebey.net> <http://www.meebey.net>
+ * Copyright (c) 2008-2009 Thomas Bruderer <apophis@apophis.ch>
+ * Copyright (c) 2015 Katy Coe <djkaty@start.no> <http://www.djkaty.com>
+ * 
+ * Full LGPL License: <http://www.gnu.org/licenses/lgpl.txt>
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+using System;
+using WebSocketSharp;
+
+namespace Meebey.SmartIrc4net
+{
+    /// <summary>
+    /// WebSocket-based IRC protocol using websocket-sharp library
+    /// </summary>
+    public class IrcWebSocketTransport : IIrcTransportManager
+    {
+        private bool _IsConnected;
+        private bool _IsConnectionError;
+
+        /// <summary>
+        /// Event which fires when a line of text is received from the connection
+        /// </summary>
+        public event ReadLineEventHandler OnMessageReceived;
+
+#pragma warning disable CS3003 
+        /// <summary>
+        /// The underlying WebSocket of the connection
+        /// </summary>
+        public WebSocket Socket {
+            get; protected set;
+        } = null;
```

Comment:
What is this supposed to do? The code should stay source compatible with .NET 3.5, not sure if I bumped Smuxi to .NET 4.0 already...